src/main/java/com/blackducksoftware/integration/fortify/batch/step/BlackDuckFortifyPhoneHomeStep.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (C) 2017 Black Duck Software Inc.
+ * http://www.blackducksoftware.com/
+ * All rights reserved.
+ *
+ * This software is the confidential and proprietary information of
+ * Black Duck Software ("Confidential Information"). You shall not
+ * disclose such Confidential Information and shall use it only in
+ * accordance with the terms of the license agreement you entered into
+ * with Black Duck Software.
+ */
+package com.blackducksoftware.integration.fortify.batch.step;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+import org.apache.log4j.Logger;
+import org.springframework.batch.core.ExitStatus;
+import org.springframework.batch.core.StepContribution;
+import org.springframework.batch.core.StepExecution;
+import org.springframework.batch.core.StepExecutionListener;
+import org.springframework.batch.core.scope.context.ChunkContext;
+import org.springframework.batch.core.step.tasklet.Tasklet;
+import org.springframework.batch.repeat.RepeatStatus;
+
+import com.blackducksoftware.integration.fortify.batch.util.HubServices;
+import com.blackducksoftware.integration.fortify.batch.util.PropertyConstants;
+import com.blackducksoftware.integration.hub.dataservice.phonehome.PhoneHomeDataService;
+import com.blackducksoftware.integration.phonehome.PhoneHomeRequestBodyBuilder;
+import com.blackducksoftware.integration.phonehome.enums.PhoneHomeSource;
+import com.blackducksoftware.integration.phonehome.enums.ThirdPartyName;
+
+/**
+ * @author jfisher
+ *
+ */
+public class BlackDuckFortifyPhoneHomeStep implements Tasklet, StepExecutionListener {
+    private final static Logger logger = Logger.getLogger(BlackDuckFortifyPhoneHomeStep.class);
+
+    private String startJobTimeStamp;
+
+    private HubServices hubServices;
```

Comment:
Should these two be final?

Reply:
Actually, wouldn't I want to make all three (startJobTimeStamp, hubServices, and propertyConstants) final?

Reply:
Ideally, yes.

Reply:
I was able to make `hubServices` and `propertyConstants` `final`. I could not make `startJobTimeStamp` as that is updated by the overridden `beforeStep` method.